### PR TITLE
ppd-emit.c: Fix SEGV in 'ppdEmitString()'

### LINKED ISSUE #849: SEGV ppd-emit.c:905:22 in ppdEmitString()
**Describe the bug**
I have found SIGSEGV crashes with cups upstream version (79c602c), using unit test `testppd.c` as a harness for fuzzing.
Here is the test.ppd file, which causes SEGV:
[testppd.tar.gz](https://github.com/OpenPrinting/cups/files/13778319/testppd.tar.gz)

This problem arises due to the lack of verification of size structure for NULL value in `ppdEmitString()`. In that case function `ppdPageSize()` returns NULL, which is recorded into structure size, then there is an appeal to the element lenght of size structure, which causes SEGV.

https://github.com/OpenPrinting/cups/blob/79c602c60e53af4a1d487d9a678c77b179ed1e4f/cups/ppd-emit.c#L891

https://github.com/OpenPrinting/cups/blob/79c602c60e53af4a1d487d9a678c77b179ed1e4f/cups/ppd-emit.c#L905

**To Reproduce**
1. move given test.ppd file in cups/ directory
2. build project with clang-13 and ASAN

```shell
export CC=clang-13
export CXX=clang++-13
export CFLAGS="-fPIC -fprofile-instr-generate -fcoverage-mapping -fsanitize=address -g"
export CXXFLAGS="-fPIC -fprofile-instr-generate -fcoverage-mapping -fsanitize=address -g"
export LDFLAGS="-g -fsanitize=address"
./configure --enable-static --disable-shared
make
```
3. build testppd target

```shell
cd cups
make testppd
```
4. run testppd 

```shell
./testppd
AddressSanitizer:DEADLYSIGNAL
=================================================================
==510142==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000030 (pc 0x561e581ec51f bp 0x7ffcb1f1d3b0 sp 0x7ffcb1f1cb20 T0)
==510142==The signal is caused by a READ memory access.
==510142==Hint: address points to the zero page.
    #0 0x561e581ec51f in ppdEmitString /home/as/cups-oss/cups-opensource/cups/ppd-emit.c:905:22
    #1 0x561e58114a8c in main /home/as/cups-oss/cups-opensource/cups/testppd.c:441:14
    #2 0x7fb9ea8461c9 in __libc_start_call_main csu/../sysdeps/nptl/libc_start_call_main.h:58:16
    #3 0x7fb9ea846284 in __libc_start_main csu/../csu/libc-start.c:360:3
    #4 0x561e58065080 in _start (/home/as/cups-oss/cups-opensource/cups/testppd+0xbb080)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/as/cups-oss/cups-opensource/cups/ppd-emit.c:905:22 in ppdEmitString
==510142==ABORTING
```

**Expected behaviour**
Test case doesn't crash.

**System Information:**
- OS and its version: debian, 12
- CUPS version: upstream (79c602c)

### PR DESCRIPTION
**Pull request by Drawishe**
When using `testppd.c` as a harness, a fuzzer found a way to call `ppdPageSize()` with NULL return value.
This caused a segmentation fault because the size structure, which is used by `values[pos]`, was assigned a NULL value.
To avoid this, we need to add a NULL value check for the size structure, free allocated memory, and return NULL.

Fixes #849

### CODE REVIEW

File: cups/ppd-emit.c [hunk 1]
```diff
@@ -888,7 +888,12 @@ ppdEmitString(ppd_file_t    *ppd,	/* I - PPD file record */
         cupsCopyString(bufptr, "%%BeginFeature: *CustomPageSize True\n", (size_t)(bufend - bufptr + 1));
         bufptr += 37;
 
-        size = ppdPageSize(ppd, "Custom");
+        if ((size = ppdPageSize(ppd, "Custom")) == NULL)
+        {
+          free(buffer);
+          free(choices);
+          return(NULL);
```
Comment (zdohnal) [inline cups/ppd-emit.c:895]: Looks good in general, only a code style issue - please put a space between return and (NULL);.

**Review by zdohnal — CHANGES_REQUESTED** [review]

**Comment by zdohnal** [comment]
Note to myself - push it into libppd once merged.

**Review by zdohnal — APPROVED** [review]
LGTM, thank you Kirill!

**Comment by zdohnal** [comment]
@Drawishe thank you for the fix!

**Comment by zdohnal** [comment]
One late note - it would be great if you stashed the commits next time if possible, it gives cleaner git history.

**Comment by jsmeix** [comment]
@zdohnal 
did you really mean "git stash"?

**Comment by zdohnal** [comment]
@jsmeix thanks Johannes, it was really "squash"....

**Pull request merged**